atlasexec/atlas.go

```diff
@@ -319,17 +319,23 @@ func (c *Client) MigrateLint(ctx context.Context, params *MigrateLintParams) (*S
 		return nil, errors.New("atlasexec: Writer or Web reporting are not supported with MigrateLint, use MigrateLintError")
 	}
 	r, err := c.runCommand(ctx, lintArgs(params))
+	if cliErr := (cliError{}); errors.As(err, &cliErr) && cliErr.summary == "" {
+		r = strings.NewReader(cliErr.detail)
+		err = nil
+	}
 	return jsonDecode[SummaryReport](r, err)
 }
 
 // MigrateLintError runs the 'migrate lint' command, the output is written to params.Writer
 func (c *Client) MigrateLintError(ctx context.Context, params *MigrateLintParams) error {
 	r, err := c.runCommand(ctx, lintArgs(params))
-	if err != nil {
-		return err
+	if cliErr := (cliError{}); errors.As(err, &cliErr) && cliErr.summary == "" {
+		r = strings.NewReader(cliErr.detail)
 	}
-	if params.Writer != nil {
-		_, err = io.Copy(params.Writer, r)
+	if params.Writer != nil && r != nil {
+		if _, ioErr := io.Copy(params.Writer, r); ioErr != nil {
+			err = errors.Join(err, ioErr)
+		}
 	}
```

Comment:
`cliError` can have stdout and stderr. These summary/details are confusing. 

Then, in the linting code, copy stdout to the writer, and return the error.